File: APISolidarityManager/APISolidarityManager/Entities/Family.cs
namespace APISolidarityManager.Models
{
    public class Family
    {
        public Guid Id { get; set; }

        public string ResponsibleName { get; set; } = null!;

        public string? ResponsibleDocument { get; set; }

        public string? PhoneNumber { get; set; }

        public string? Address { get; set; }

        public decimal? MonthlyIncome { get; set; }

        public string? Notes { get; set; }

        public bool Active { get; set; } = true;

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public ICollection<FamilyMember> FamilyMembers { get; set; } = new List<FamilyMember>();

        public ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();
    }
}

File: donation-system-web/src/types/family.ts
export interface FamilyResponse {
  id: string;
  responsibleName: string;
  responsibleDocument?: string;
  phoneNumber?: string;
  address?: string;
  monthlyIncome?: number;
  notes?: string;
  active: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface FamilyPriorityResponse {
  familyId: string;
  priorityScore: number;
  priorityLevel: string;
  requiresManualAnalysis: boolean;
  reasons: string[];
}

export interface FamilyPriorityListItemResponse {
  familyId: string;
  responsibleName: string;
  priorityScore: number;
  priorityLevel: string;
  requiresManualAnalysis: boolean;
  reasons: string[];
  lastDeliveryDate?: string;
}

export interface UpdateFamilyRequest {
  responsibleName: string;
  responsibleDocument?: string;
  phoneNumber?: string;
  address?: string;
  monthlyIncome?: number;
  notes?: string;
  active: boolean;
}

export interface CreateFamilyRequest {
  responsibleName: string;
  responsibleDocument?: string;
  phoneNumber?: string;
  address?: string;
  monthlyIncome?: number;
  notes?: string;
}
